src/ScheduleFunctions.cpp

```diff
@@ -1855,6 +1858,33 @@ bool validate_schedule(Function f, Stmt s, const Target &target, bool is_output,
                 }
             }
         }
+
+        // Check that extern stages do not have any non-extern loops
+        // inside any extern loops, and all loop types are supported
+        // for extern stages.
+        const vector<Dim>& dims = f.definition().schedule().dims();
+        bool is_extern = !dims.empty() ? dims.front().for_type == ForType::Extern : false;
+        for (const Dim& i : dims) {
+            switch (i.for_type) {
+            case ForType::Extern:
+                if (!is_extern) {
+                  user_error
+                      << "Externally defined Func " << f.name()
+                      << " cannot have extern loop " << i.var
+                      << " outside a non-extern loop.\n";
+                }
+                break;
+            case ForType::Serial:
+            case ForType::Parallel:
+            case ForType::Unrolled:
+                is_extern = false;
+                break;
+            default:
+                user_error
+                    << "Externally defined Func " << f.name()
```

Comment:
GPUBlocks is a pretty interesting type of loop here to allow in the future. It would be used to call external code that operates across an entire block (e.g. a parallel reduction) on a GPU as the output to a kernel.